tcms/rpc/api/testcase.py

```diff
@@ -304,9 +314,20 @@ def filter(query=None):  # pylint: disable=redefined-builtin
             "default_tester__username",
             "reviewer",
             "reviewer__username",
+            "setup_duration",
+            "testing_duration",
+            "expected_duration",
         )
         .distinct()
     )
+    result = [testcase_dict for testcase_dict in qs.iterator()]
+
+    for testcase_dict in result:
+        for duration_key in ["setup_duration", "testing_duration", "expected_duration"]:
+            if testcase_dict[duration_key] is not None:
+                testcase_dict[duration_key] = str(testcase_dict[duration_key])
```

Comment:
This is sub-optimal for large queries. Will using `Cast` work to cast those fields to string directly via SQL ?

Reply:
Yes. I can Cast them to string by annotating the filtered queryser appropriately.

Although, that would mean the items in `result` won't contain the keys `setup_duration`, `testing_duration`, etc. as querysets cannot be annotated with already existing field names.

I'll make this change later in the day when I'm on my PC.

Reply:
> Although, that would mean the items in result won't contain the keys setup_duration, testing_duration, etc. as querysets cannot be annotated with already existing field names.

Do you have a reference for this? If yes add it as a comment in the source. 

Prefix with `_` if possible so we can see how it works. Maybe it will be an acceptable compromise.